### PR TITLE
History api modifications

### PR DESCRIPTION
**Pull request by maheshsattala**
Modified history api with from_date and to_date in place of month.

### CODE REVIEW

File: kairon/api/app/routers/history.py [hunk 1]
```diff
@@ -158,75 +196,74 @@ async def engaged_users_trend(month: int = Query(default=6, ge=1, le=6),
...
+async def retention_trend(from_date: datetime.date = datetime.date.today() - datetime.timedelta(30),
+                          to_date: datetime.date = datetime.date.today(),
+                          current_user: User = Security(Authentication.get_current_user_and_bot, scopes=TESTER_ACCESS)):
     """
     Fetches the counts of user retention percentages of the bot for previous months
     """
     return Utility.trigger_history_server_request(
         current_user.get_bot(),
         f'/api/history/{current_user.get_bot()}/trends/users/retention',
-        {'month': month}
```
Comment (udit-pandey) [inline kairon/api/app/routers/history.py:198]: why this api has been removed?

File: tests/integration_test/history_services_test.py [hunk 2]
```diff
@@ -105,279 +105,279 @@ def test_chat_history_with_user_id_contains_special_character(mock_chat_history)
 
 
 @mock.patch('kairon.history.processor.MongoClient', autospec=True)
-def test_visitor_hit_fallback(mock_mongo):
```
Comment (udit-pandey) [inline tests/integration_test/history_services_test.py:108]: lot of endpoint changes have been made to tests. we were required to only change the data format to date range wherever month was required. why so many lines of change?

File: kairon/api/app/routers/history.py [hunk 3]
```diff
@@ -308,81 +360,101 @@ async def word_cloud(
...
+        to_date: datetime.date = datetime.date.today(),
+        current_user: User = Security(Authentication.get_current_user_and_bot, scopes=TESTER_ACCESS)):
     """
     Fetches the total session count for users for the past months.
     """
     return Utility.trigger_history_server_request(
         current_user.get_bot(),
         f'/api/history/{current_user.get_bot()}/metrics/sessions/total',
-        {'month': month}
+        {'from_date': from_date.strftime("%d-%m-%Y"), 'to_date': to_date.strftime("%d-%m-%Y")}
```
Comment (udit-pandey) [inline kairon/api/app/routers/history.py:457]: make a separate utility to convert date to string

**Review by udit-pandey — CHANGES_REQUESTED** [review]

File: kairon/api/app/routers/history.py (hunk 1 above)
Comment (maheshsattala) [reply kairon/api/app/routers/history.py:198]: I haven't removed it. I just moved it to line no 54.

File: tests/integration_test/history_services_test.py (hunk 2 above)
Comment (maheshsattala) [reply tests/integration_test/history_services_test.py:108]: I didn't change the endpoints. I moved a few test cases to the top.

File: kairon/api/app/routers/history.py (hunk 3 above)
Comment (maheshsattala) [reply kairon/api/app/routers/history.py:457]: I have written separate utility function and made changes accordingly

File: kairon/api/app/routers/history.py [hunk 4]
```diff
@@ -21,212 +21,278 @@
 
 
 @router.get("/users", response_model=Response)
-async def chat_history_users(month: int = Query(default=1, ge=1, le=6), current_user: User = Security(Authentication.get_current_user_and_bot, scopes=TESTER_ACCESS)):
+async def chat_history_users(
+        from_date: datetime.date = datetime.date.today() - datetime.timedelta(30),
```
Comment (udit-pandey) [inline kairon/api/app/routers/history.py:25]: will date.today() return value in utc?
if not, shouldnt we be using datetime.utcnow().date() instead?

File: kairon/api/app/routers/history.py (hunk 1 above)
Comment (udit-pandey) [reply kairon/api/app/routers/history.py:198]: why?

File: tests/integration_test/history_client_test.py [hunk 5]
```diff
@@ -483,146 +583,118 @@ def test_engaged_user_range_with_kairon_client(mock_auth, mock_mongo_processor):
 
 
 @responses.activate
-def test_new_user_range_with_kairon_client(mock_auth, mock_mongo_processor):
```
Comment (udit-pandey) [inline tests/integration_test/history_client_test.py:486]: why were these test cases removed?

File: tests/integration_test/history_client_test.py [hunk 6]
```diff
@@ -483,146 +583,118 @@ def test_engaged_user_range_with_kairon_client(mock_auth, mock_mongo_processor):
...
-
-
-@responses.activate
-def test_user_retention_range_with_kairon_client(mock_auth, mock_mongo_processor):
+def test_engaged_user_range_with_value_with_kairon_client(mock_auth, mock_mongo_processor):
+    from_date = datetime.date.today() - datetime.timedelta(30)
+    to_date = datetime.date.today()
     responses.add(
         responses.GET,
-        f"https://localhost:8083/api/history/{pytest.bot}/trends/users/retention",
```
Comment (udit-pandey) [inline tests/integration_test/history_client_test.py:534]: why endpoint changed in rest of the test cases?
the test method names have also been changed. why?

File: tests/integration_test/history_services_test.py (hunk 2 above)
Comment (udit-pandey) [reply tests/integration_test/history_services_test.py:108]: why? what was the need?

File: tests/integration_test/history_services_test.py [hunk 7]
```diff
@@ -443,78 +443,34 @@ def test_download_conversation_with_error(monkeypatch):
 
 
 @mock.patch('kairon.history.processor.MongoClient', autospec=True)
-def test_chat_history_no_token(mock_mongo):
```
Comment (udit-pandey) [inline tests/integration_test/history_services_test.py:446]: why test cases removed?

File: tests/integration_test/history_services_test.py [hunk 8]
```diff
@@ -443,78 +443,34 @@ def test_download_conversation_with_error(monkeypatch):
...
 
     actual = response.json()
     assert actual["error_code"] == 0
-    assert len(actual["data"]["history"]) == 0
+    assert actual["data"]["total_conversation_range"] == {}
     assert actual["message"] is None
     assert actual["success"]
 
 
-def test_no_bot_id():
```
Comment (udit-pandey) [inline tests/integration_test/history_services_test.py:489]: why test case removed?

**Review by udit-pandey — CHANGES_REQUESTED** [review]

**Review by udit-pandey — APPROVED** [review]
approved

**Pull request merged**